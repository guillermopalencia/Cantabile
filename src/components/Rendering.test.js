import { getByRole, render, screen, cleanup } from '@testing-library/react';
import renderer  from 'react-test-renderer'
import TestThis from './TestComponent';

afterEach(() => {
    cleanup()
});

describe("Test Component Render Test", () =>{
    it("should render Test component", () => {
      const { getByRole } = render(<TestThis />)
      expect(getByRole('heading', {hidden: true})).toBeInTheDocument();
    })

    it("should find input fields", () =>{
      render(<TestThis />)
      const element = screen.getByRole
      expect(element('textbox', {hidden: true})).toBeInTheDocument()
    })
  })

describe("Snapshot testing", () => {
  it("Should match exisiting snapshot", () => {
    const element = renderer.create(<TestThis />)

    let checker = element.toJSON
    expect(checker).toMatchSnapshot()
  })
})