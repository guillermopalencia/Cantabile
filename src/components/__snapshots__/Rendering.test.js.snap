// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Should match exisiting snapshot 1`] = `[Function]`;
